slapp/lambdas/post_annotation.py

```diff
@@ -1,42 +1,106 @@
 import json
 import boto3
 from urllib.parse import urlparse
+import math
 
 
 def lambda_handler(event, context) -> dict:
     """
     The lambda function to be run upon the completion of a labeling task
     by a user. Transports the data to an s3 bucket for later use or movement to
     the allen server.
-    Args:
-        event: dict structure containing input data passed from label task to
-        lambda function
-        context: context of the event
-    Returns: bool status of uploading to bucket
+    Parameters
+    ----------
+        event: dict
+            Json-formatted event from Sagemaker Ground Truth
+            Event doc: https://docs.aws.amazon.com/sagemaker/latest/dg/sms-custom-templates-step3.html    # noqa
+        context: object
+            Lambda Context runtime methods and attributes
+            Context doc: https://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html    #noqa
+
+    Returns
+    -------
+        Json-formatted list of consolidated labels, with majority
+        label calculated if it exists.
+        [
+            {
+                "datasetObjectId": <id>,
+                "consolidatedAnnotation": {
+                    "content" : {
+                        <labelAttributeName> : {
+                            "sourceData": <source>,
+                            "majorityLabel": <majority label>,
+                            "workerAnnotations": [
+                                {"workerId": <worker_id>, "roiLabel": <label>},
+                                ...
+                            ]
+                        }
+                    }
+                }
+            },
+            ...
+        ]
+        Return doc: https://docs.aws.amazon.com/sagemaker/latest/dg/sms-custom-templates-step3.html    # noqa
     """
+    # For CloudWatch Logs
+    print("#### EVENT ####")
```

Comment:
how about this instead?
https://docs.aws.amazon.com/lambda/latest/dg/python-logging.html#python-logging-lib

Reply:
I typically use logging, but messages to stdout get nicely captured in CloudWatch logs like this.
<img width="1229" alt="Screen Shot 2020-05-07 at 11 39 14 AM" src="https://user-images.githubusercontent.com/34227334/81332268-a4d1f880-9057-11ea-9b91-94a6b46b6016.png">
I want to double check that stderr logs are captured too before switching.

Reply:
Checked it, looks like the logs are captured when I used a Test Event in the console:

```
Function Logs:
START RequestId: 26c6fd74-36a8-44ee-892d-85b00a458fa7 Version: $LATEST
#### EVENT ####
{"version": "2018-10-06", "labelingJobArn": "arn:aws:sagemaker:us-east-1:111111111111:labeling-job/test-labeling-job", "payload": {"s3Uri": "s3://dev.slapp.alleninstitute.org/test_lambda/test_consolidation_payload.json"}, "labelAttributeName": "test-label-job", "roleArn": "arn:aws:iam::606907419058:role/gt-lambdas-LambdaExecutionRole-1BADDIO7S6NMP", "outputConfig": "s3://dev.slapp.alleninstitute.org/test_lambda/outputs"}
### CONSOLIDATED LABELS ###
[{"datasetObjectId": "1", "consolidatedAnnotation": {"content": {"test-label-job": {"sourceData": "s3://dev.slapp.alleninstitute.org/UIUX/20200504160215/outline_1574684.png", "majorityLabel": "cell", "workerAnnotations": [{"workerId": "private.us-west-2.c3910846ae319e37", "roiLabel": "not cell"}, {"workerId": "private.us-west-2.926e79f24e7be00f", "roiLabel": "cell"}, {"workerId": "private.us-west-2.bf5e10f035776790", "roiLabel": "cell"}, {"workerId": "private.us-west-2.88cb406cec4abd35", "roiLabel": "cell"}]}}}}, {"datasetObjectId": "3", "consolidatedAnnotation": {"content": {"test-label-job": {"sourceData": "s3://dev.slapp.alleninstitute.org/UIUX/20200504160215/outline_1574704.png", "majorityLabel": "not cell", "workerAnnotations": [{"workerId": "private.us-west-2.c3910846ae319e37", "roiLabel": "not cell"}, {"workerId": "private.us-west-2.bf5e10f035776790", "roiLabel": "not cell"}, {"workerId": "private.us-west-2.926e79f24e7be00f", "roiLabel": "not cell"}, {"workerId": "private.us-west-2.88cb406cec4abd35", "roiLabel": "not cell"}]}}}}]
END RequestId: 26c6fd74-36a8-44ee-892d-85b00a458fa7
REPORT RequestId: 26c6fd74-36a8-44ee-892d-85b00a458fa7	Duration: 280.68 ms	Billed Duration: 300 ms	Memory Size: 128 MB	Max Memory Used: 76 MB	
```

Updated.